PREVIEW-FEATURES.md

```diff
@@ -0,0 +1,45 @@
+# Preview features
+
+## Introduction
+
+Preview features are features whose design, specification, and implementation are complete, but which
+would benefit from a period of broad exposure and evaluation before either achieving final and permanent
+status in JavaFX or else being refined or removed.
+
+A feature should meet two criteria in order to be considered complete:
+
+1. **Readiness**: The feature should be finished in the next release of JavaFX.
+   APIs with an exceptionally large surface area, or features that deeply affect the semantics or usage
+   of the JavaFX platform, may require additional rounds of feedback and revision.
+2. **Stability**: The feature could credibly achieve final and permanent status with no further changes.
+   This implies an extremely high degree of confidence in the concepts which underpin the feature, but
+   does not completely rule out making syntactic or semantic changes in response to feedback.
+
+The key properties of a preview feature are:
+
+1. **High quality**: A preview feature must display the same level of technical excellence and finesse as
+   a final and permanent feature of JavaFX. For example, a preview feature must respect traditional Java
+   principles such as readability and compatibility.
+
+2. **Not experimental**: A preview feature must not be experimental, risky, incomplete, or unstable.
+   For the purpose of comparison, if an experimental feature is considered 25% "done", then a preview
+   feature should be at least 90% "done".
+
+3. **Opt-in:** The use of preview features must be explicitly enabled by application developers by
+   setting a system property. The implementation must detect the opt-in, and fail at runtime when the
+   application has not opted into the use of preview features. All preview features have equal status
+   in any given JavaFX release and can not be enabled individually.
+
```

Comment:
do you want to talk about similarities and differences between this and jdk preview features https://openjdk.org/jeps/12 ?

Reply:
This definition is, as you point out, largely lifted from JEP 12. Aside from wording changes, here are some key differences:

* JEP 12 assumes that a feature should be finished within two releases. I think for JavaFX, being just a framework and not a programming language, it is sensible to assume that we can finish features in the next release; that is, a preview feature that is introduced in JavaFX N should be finished in N+1.
* JEP 12 says that a preview feature should be at least 95% "done". I think we can be a little bit more risk-tolerant here, which is why I lowered this number to 90% "done".

The general idea is that preview features in JavaFX should be faster and accept a little bit more risk than preview features in the JDK.

Reply:
The space of people who know what a JDK preview is is larger than that of JFX people.  I think we might want to point to JEP12 and explicitly explain the difference, that's all.

Reply:
We could also do that, but the problem is that the contents of JEP 12 only map to JavaFX in some parts, but not in others. So we would need to include weasel language like "where applicable" (which always leaves a bit of room for interpretation), or provide what would basically be a detailed sentence-by-sentence diff.

So in the end, it might still be easier to point at our own (complete) definition instead of a partially applicable definition somewhere else.

Reply:
> So in the end, it might still be easier to point at our own (complete) definition instead of a partially applicable definition somewhere else.

Even if this ends up being the way we go, a pointer to JEP 12 and list of key differences would be helpful.